reference/extensions/hooks.rst

```diff
@@ -1,4 +1,167 @@
 .. _reference_extensions_hooks:
 
 Hooks
------
\ No newline at end of file
+-----
+
+The Conan hooks is a feature intended to extend the Conan functionalities and let users customize the client behavior at determined
+points.
+
+Hook structure
+--------------
+
+A hook is a Python function that will be executed at certain points of Conan workflow
+to customize the client behavior without modifying the client sources or the recipe ones.
+In the :ref:`hooks reference <hooks_reference>` you can find the full list of hook functions
+and exhaustive documentation about their arguments.
+
+Hooks can implement any functionality: it could be Conan commands, recipe interactions
+such as exporting or packaging, or interactions with the remotes.
+
+Here is an example of a simple hook:
+
+.. code-block:: python
+   :caption: *hook_example.py*
+
+    import os
+    from conan.tools.files import load
+
+    def pre_export(conanfile):
+        for field in ["url", "license", "description"]:
+            field_value = getattr(conanfile, field, None)
+            if not field_value:
+                conanfile.output.error(f"[REQUIRED ATTRIBUTES] Conanfile doesn't have '{field}'.
+                                          It is recommended to add it as attribute.")
+
+    def pre_source(conanfile):
+        conanfile_content = load(conanfile, os.path.join(conanfile.recipe_folder, "conanfile.py"))
+        if "def source(self):" in conanfile_content:
+            test = ""
+            valid_content = [".zip", ".tar", ".tgz", ".tbz2", ".txz"]
+            invalid_content = ["git checkout master", "git checkout devel", "git checkout develop"]
+            if "git clone" in conanfile_content and "git checkout" in conanfile_content:
+                fixed_sources = True
+                for invalid in invalid_content:
+                    if invalid in conanfile_content:
+                        fixed_sources = False
+            else:
+                fixed_sources = False
+                for valid in valid_content:
+                    if valid in conanfile_content:
+                        fixed_sources = True
+
+            if not fixed_sources:
+                output.error(f"[IMMUTABLE SOURCES] Source files does not come from and immutable place. Checkout to a "
+                            "commit/tag or download a compressed source file for {conanfile.ref}")
```

Comment:
All of this sounds like an example, not reference material. I am fine with keeping a very simple, minimal example here im the reference, but the "immutable sources" seems too much.

Reply:
removed the second example.